src/odemis/odemisd/mdupdater.py

```diff
@@ -178,23 +182,30 @@ def observeLens(self, lens, comp):
         # MD_LENS_MAG.
         # For DigitalCamera, the .pixelSize is the SENSOR_PIXEL_SIZE, and we
         # compute PIXEL_SIZE every time the LENS_MAG *or* binning change.
-        if model.hasVA(comp, "scale"):
+        if model.hasVA(comp_affected, "scale"):
             md_va_list["magnification"] = model.MD_LENS_MAG
         else:
             # TODO: instead of updating PIXEL_SIZE everytime the CCD changes binning,
             # just let the CCD component compute the value based on its sensor
             # pixel size + MAG, like for the scanners.
-            if model.hasVA(comp, "binning"):
-                binva = comp.binning
+            if model.hasVA(comp_affected, "binning"):
+                binva = comp_affected.binning
             else:
                 logging.debug("No binning")
                 binva = None
 
             # Depends on the actual size of the ccd's density (should be constant)
-            captor_mpp = comp.pixelSize.value  # m, m
+            captor_mpp = comp_affected.pixelSize.value  # m, m
 
             # we need to keep the information on the detector to update
-            def updatePixelDensity(unused, lens=lens, comp=comp, binva=binva):
+            def updatePixelDensity(unused, lens=lens, comp_affected=comp_affected, binva=binva):
+                # Note: The scope of variables is redefined in the nested/local function
+                # to ensure that the correct variables are used and were not overwritten
+                # before calling the function.
+                # unused = mag: unused as mag might have also been overwritten
```

Comment:
No. Don't mix up: the unused here, is just because the function is called both when the magnification is changed *and* when binva is changed. So the first argument contains either mag, or binning, depending on the case.
In any case, only put this comment once. At the top of the first function is enough.

Reply:
true: something like: 
arg "unused": magnification or binning (function called when mag and/or binning changes) 
or put nothing at all?